cifopy/charles/selection.py

```diff
@@ -8,14 +9,26 @@ def fps(population):
 
     Args:
         population (Population): The population we want to select from.
+        prem: By default False, in case the variance of the fitness distribution in the population is less than
+        5% of the initial variance
 
     Returns:
         Individual: selected individual.
     """
 
     if population.optim == "max":
         # Sum total fitnesses
-        total_fitness = sum([i.fitness for i in population])
+        fitness = [i.fitness for i in population]
+        """
+        if prem:
+            #Do we want to save this variance? Maybe we can plot how it changes through generation
+            var = np.var(fitness)
+            if var/population.initial_var<0.05:
+                print("Premature convergence! Reshuffling the deck")
+                population.reshuffle()
+                #Re-evaluate fitness since population changed
+                fitness = [i.fitness for i in population]"""
```

Comment:
Can we delete this?

Reply:
Yes for sure, it was the previous implementation. Forgot to delete it :)